Makefile

```diff
@@ -10,6 +10,10 @@ unit_test:
 fix:
 	pipenv run pytest -x tests
 
+.PHONY: integration_test
+integration_test:
+	cd tests/integration && export PIPENV_PIPFILE=../../../Pipfile && for integration_test in *; do cd "$${integration_test}"; pipenv run sh runme.sh opera; cd ..; done
```

Comment:
This line has quite a few problems ;)

Now, every time we see the loop in the Makefile, we should think again about our implementation because make works best if we only specify targets and their dependencies, and then let make do the looping. My suggestion would be the following:

```
# List of all integration test scenarios
integration_test_scenarios := $(wildcard tests/integration/*/runme.sh)

# Default opera executable (tailored for developers). Can be overridden by
# setting the OPERA environment variable to something else.
OPERA ?= "pipenv run opera"

# Poitn pipenv to the right file since we can execute tests from a subdir.
export PIPENV_PIPFILE := $(realpath Pipfile)

.PHONY: integration_test
integration_test: $(integration_test_scenarios)

.PHONY: $(integration_test_scenarios)
$(integration_test_scenarios):
	cd $(dir $@) && bash $(notdir $@) $(OPERA)
```

Implementing things this way also removes the hardcoded test command that we had in the Makefile before.

Reply:
Yes as I thought my command is not the best thing here. An wow, your proposed solution seems really cool and promising so I will definitely include this.